### DISCUSSION TITLE
Accept collection based offers

### DISCUSSION THREAD

**Discussion by mustafa222lk** [ProjectOpenSea/seaport#969] [Q&A]
Hey there,
I am trying to accept a collection based offer for my nft. Im retrieveing the offers from the opensea api endpoint and he choosing a offer randomly. I succeed when it comes to single nft absed offers but when i try to accept a collection based offer it doesnt work.

`        const { executeAllActions: fulfillOrder } = await spClient.fulfillOrder({
            order,
            accountAddress: this.walletAddress,
            considerationCriteria: [{ identifier: tokenId, proof: new MerkleTree([tokenId]).getProof(tokenId, 1) }]
          });
`
This is my code. Im not sure if im generating a wrong merkle proof or not. 

Getting this error:
: invalid arrayify value (argument="value", value={"position":"left","data":{"type":"Buffer","data":[71,71]}}, code=INVALID_ARGUMENT, version=bytes/5.6.1)

What am i doing wrong ?

**Comment by ryanio** [comment]
You need the info from this endpoint to fulfill an offer: https://docs.opensea.io/reference/fulfill-an-offer
